cdap-app-templates/cdap-etl/cdap-etl-realtime/src/main/java/co/cask/cdap/templates/etl/realtime/ETLRealtimeTemplate.java

```diff
@@ -52,6 +54,8 @@ public void configureAdapter(String adapterName, ETLRealtimeConfig etlConfig, Ad
     super.configureAdapter(adapterName, etlConfig, configurer);
     configurer.addRuntimeArgument(Constants.CONFIG_KEY, GSON.toJson(etlConfig));
     configurer.setInstances(etlConfig.getInstances());
+    // Generate unique id for this adapter creation.
+    configurer.addRuntimeArgument(Constants.Realtime.UNIQUE_ID, UUID.randomUUID().toString());
```

Comment:
You can just use timestamp, assuming you don't create two adapter instances in the same millisecond.